python-client/giskard/slicing/filters.py

```diff
@@ -9,6 +12,35 @@ def filter(self, slices):
         raise NotImplementedError()
 
 
+class InternalTestFilter(DataSliceFilter):
+
+    def __init__(self, model, og_dataset, test=test_diff_f1, threshold=0.1):
+        self.dataset = og_dataset
+        self.model = model
+        self.test = test
+        self.threshold = threshold
+
+    def _diff_test(self, data_slice):
+        # Convert slice to Giskard Dataframe
+        g_dataset = Dataset(data_slice,
+                            target_col=self.dataset.target,
+                            class_labels=self.dataset.class_labels,
+                            column_types=self.dataset.column_types
+                            )
+        # Apply the test
+        test_res = self.test(
+            actual_slice=g_dataset,
+            reference_slice=self.dataset,
```

Comment:
Would prefer to have exclude the slice from the reference, for independence.

Reply:
Noted